cpp/src/arrow/csv/writer.cc

```diff
@@ -136,15 +134,14 @@ class ColumnPopulator {
   MemoryPool* const pool_;
 };
 
-// Copies the contents of to out properly escaping any necessary characters.
-// Returns the position prior to last copied character (out_end is decremented).
-char* EscapeReverse(arrow::util::string_view s, char* out_end) {
-  for (const char* val = s.data() + s.length() - 1; val >= s.data(); val--, out_end--) {
-    if (*val == '"') {
-      *out_end = *val;
-      out_end--;
+// Copies the contents of s to out properly escaping any necessary characters.
+// Returns the position next to last copied character.
+char* Escape(arrow::util::string_view s, char* out_end) {
+  for (const char c : s) {
```

Comment:
It occurs to me that performance here might be improved by using higher level methods to find the next escape character and memcpy in between.  Probably depends on string lengths being processed.

Reply:
Yes. It's the main reason I change the scan order. This function is the hotspot of string with quotes benchmark. Reverse scanning makes it hard to improve, the output string header pos is not known until end.